src/rgb/contract/assignments.rs

```diff
@@ -708,3 +708,726 @@ mod strict_encoding {
+        "8f75db9f89c7c75f0a54322f18cd4d557ae75c24a8e5a95eae13fe26edc2d789",
+    ];
+
+    // State data used in CustomData type Assignments
+    static STATE_DATA: [&str; 4] = [
+        "e70a36e2ce51d31d4cf5d6731fa63738648127db833715d39652d86d927d4888",
+        "408e331ebce96ca98cfb7b8a6286a79300379eed6395636e6d103017d474039f",
+        "c6411aea082e2c5d74347368677db69545126237d5ed78faa0846352f5383f95",
+        "277fb00655e2523424677686c24d90fba6b70869050ae204782e8ef0ab8049c6",
+    ];
+
+    // Confidential seals for Declarative Assignments
+    static DECLARATIVE_OUTPOINT_HASH: [&str; 4] = [
+        "58f3ea4817a12aa6f1007d5b3d24dd2940ce40f8498029e05f1dc6465b3d65b4",
+        "6b3c1bee0bd431f53e6c099890fdaf51b8556a6dcd61c6150ca055d0e1d4a524",
+        "9a17566abc006cf335fd96d8f8a4136526d85493a85ebe875abbbee19795c496",
+        "c843ac6b197ae371191264cc0e4ed18a910b5522a0bad72a24f2080c170e2053",
+    ];
+
+    // Confidential seals for Pedersan type Assignments
+    static PEDERSAN_OUTPOINT_HASH: [&str; 4] = [
+        "281543d7f791d4b4f8ef1196e436bc3286a5505f7bafd978d4af9be6f112e1b4",
+        "32d71a47d8ff6015fc58525985af7346e0802c7ad065ad79335602c7a6562ab3",
+        "68955a27e1ffde810fcfdd18697eb59aa4f7b0afde2a8193cd28184b729b5195",
+        "698c43d973bec68540e6df67137785e40be6d29def4888ada3cd7b7884b37f62",
+    ];
+
+    // Confidential seals for CustomData type Assignments
+    static HASH_OUTPOINT_HASH: [&str; 4] = [
+        "7efe71b7a37a39da798774ca6b09def9724d81303892d55cac3edb0dc8340a3a",
+        "9565d29461c863e013c26d176a9929307286963322849a1dc6c978e5c70c8d52",
+        "9b64a3024632f0517d8a608cb29902f7083eab0ac25d2827a5ef27e9a68b18f9",
+        "dc0d0d7139a3ad6010a210e5900201979a1a09047b10a877688ee5a740ae215a",
+    ];
+
+    // Generic encode-decode testing
+    #[test]
+    fn test_encoded_data() {
+        test_encode!((HASH_VARIANT, AssignmentsVariant));
+        test_encode!((PEDERSAN_VARIANT, AssignmentsVariant));
+        test_encode!((DECLARATIVE_VARIANT, AssignmentsVariant));
+    }
+
+    // Generic garbage value testing
+    #[test]
+    #[should_panic(expected = "EnumValueNotKnown")]
+    fn test_garbage_dec() {
+        test_garbage!((DECLARATIVE_VARIANT, AssignmentsVariant));
+    }
+
+    #[test]
+    #[should_panic(expected = "EnumValueNotKnown")]
+    fn test_garbage_ped() {
+        test_garbage!((PEDERSAN_VARIANT, AssignmentsVariant));
+    }
+
+    #[test]
+    #[should_panic(expected = "EnumValueNotKnown")]
+    fn test_garbage_hash() {
+        test_garbage!((HASH_VARIANT, AssignmentsVariant));
+    }
+
+    #[test]
+    #[should_panic(expected = "UnsupportedDataStructure")]
+    fn test_garbage_ped_2() {
+        let mut bytes = PEDERSAN_VARIANT.clone();
+        bytes[1] = 0x02;
+
+        AssignmentsVariant::strict_decode(&bytes[..]).unwrap();
+    }
+
+    #[test]
+    fn test_zero_balance() {
+        // Base amounts considered as inputs [2, 3, 1, 4]
+        let amounts = [2u64, 3u64, 1u64, 4u64];
```

Comment:
Can we use here some more test cases? Like put the main test as a separate fn and run it over multiple cases (including some auto-generated and large ones)?

Reply:
And it will really help if we originally will separate inputs and outputs, not just by splitting single array into parts

Reply:
> Can we use here some more test cases?

yes possibly, like creating a function to create arbitrary allocation given a set of input amounts?

>And it will really help if we originally will separate inputs and outputs, not just by splitting single array into parts

Note we are not splitting the array anymore. The entire array is taken as input. Same for the outputs, which is created from the full balanced allocation.